### ISSUE TITLE
Potentially Dangerous alternative prefix `minecraft` for all TiCon TileEntity registration

### ISSUE THREAD

**Issue by 3TUSK** [SlimeKnights/TinkersConstruct#3573]
Since MinecraftForge/MinecraftForge#4681, the id passed in for TileEntity registration is required to be a `ResourceLocation` or at least of which can be parsed as `ResourceLocation`. 

Sample log spam using Forge 14.23.4.2714 and `TConstruct-1.12.2-2.10.1.87`:
```
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.table`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.craftingstation`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.stenciltable`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.partbuilder`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.patternchest`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.partchest`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.toolstation`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.toolforge`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.smeltery_controller`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.smeltery_component`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.tank`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.faucet`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.channel`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.casting_table`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.casting_basin`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.smeltery_drain`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.seared_furnace`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
[19:15:18] [main/WARN] [FML]: Potentially Dangerous alternative prefix `minecraft` for name `tconstruct.tinker_tank`, expected `tconstruct`. This could be a intended override, but in most cases indicates a broken mod.
```

The fix should be simple (in `Util::prefix`, change `.` to `:`); however the underlying issue is the migration of old ids, which requires `DataFix`.

**Comment by KnightMiner** [comment]
Yes, this is known, but it's not a huge priority as our rule entity names are already prefixed. Its easier to leave it as is and fix it in 1.13 than to worry about possibly breaking old worlds.

**Comment by J4rr3tt5** [comment]
I'm confused. How to fix?

**Comment by KnightMiner** [comment]
There is no fix as it's not an issue. It's just a warning in the log that we took different precautions for

**Comment by KnightMiner** [comment]
Fixed in 1.16, not worth fixing in 1.12 as that requires non-trivial migrations.
